pyiron_base/job/script.py

```diff
@@ -31,6 +31,68 @@ class ScriptJob(GenericJob):
         project (ProjectHDFio): ProjectHDFio instance which points to the HDF5 file the job is stored in
         job_name (str): name of the job, which has to be unique within the project
 
+    Example:
+        To submit a notebook as a ScriptJob job to the pyiron job management system,
+
+            1. Create the notebook to be submitted, for ex. 'example.ipynb', and save it -- Can contain any code like:
+                    ```
+                    from pyiron import Project
+                    pr = Project('example')
+                    job = pr.create_job(job_type=pr.job_type.Lammps, job_name='job')  # we name the job 'job'
+                    job.structure = pr.create_ase_bulk('Fe')  # specify structure
+
+                    # Optional: get an input value from 'submit_example_job.ipynb', the notebook which submits
```

Comment:
After reading the discussion, I think I come down on the side that the external input method is making this docstring too complex. I'd rather drop these three lines. (Keep the core specification, but just make it a number)

Reply:
The problem with not having these lines here is that there's no way for the user to know that such a functionality exists (or at least I don't know how to look it up without leaving the notebook). And I'm pretty sure that many people want to pass input arguments from the submitter to the job notebook. So I'd say making them google it is bigger evil than having three lines here.

Reply:
I would also side with @samwaseda on this. As a user, I wouldn't know if this functionality existed unless specified in some example notebook. But since we have a 'docstring version' of the notebook here in the docstring of ScriptJob itself, I don't see a better way of advertising `pr.get_external_input()` within the context of submitting notebooks.

Reply:
This would still be in a sphinx-parsed docstring, so just make a reference like this
```
See :method:`~.Project.get_external_input()` for further reading how to pass input to a notebook.
```

Reply:
I like this even less now.



> The problem with not having these lines here is that there's no way for the user to know that such a functionality exists (or at least I don't know how to look it up without leaving the notebook). 

If this is really the case, then there is a fundamental problem with `Project.get_external_output` and it should be changed. 

What is the actual intended use of this method? Is it *only* for `ScriptJob` or is there another application I'm missing? It's a public method on our base-level `Project` object; if it's only application is for `ScriptJob` then we really need to figure out a way to relocate it, or make a special `ScriptProject` or something.

I'm extremely uncomfortable with the idea that the only way to get meaningfully informed about a `pyiron_base.Project` method is by looking at the docstring of `ScriptJob`. That smells to high heaven.

Reply:
Currently, it occurs only in `ScriptJob` that we need a project level external input.

Actually I don't really understand what troubles you so much.

Reply:
Agree with @liamhuber, since it's a ScriptJob specific feature it should live in there as well, but I'm not sure where to put it.  The problem is that the notebook running in the ScriptJob does not have a reference to itself, but only one to the Project.

Reply:
It's *currently* a `ScriptJob` specific feature. Since it's conceptually not necessary elsewhere, I don't see a problem with putting it only in this example here.

Reply:
`pyiron_base.Project` is sacrosanct because it's the most fundamental object in the user hierarchy. Thus, decorating it with a public method that only applies to one specific job type is undesirable. However, I share @pmrv's concern that I don't know where else to put it right now.

Let's leave the documentation as you have it, and discuss alternative tricks in the pyiron meeting.

Reply:
As @liamhuber suggested, I'm perfectly fine with talking about it at the next pyiron meeting. I just want to say I think `get_external_input` does what I think it should, since I can totally imagine that we're gonna have project level external input in the future. So even if it's not used by `ScriptJob`, I'd like it to be there as it is now.